packages/@lwc/compiler/src/options.ts

```diff
@@ -54,13 +54,15 @@ export interface TransformOptions {
     experimentalDynamicComponent?: DynamicComponentConfig;
     outputConfig?: OutputConfig;
     isExplicitImport?: boolean;
+    experimentalLightDOMComponent?: boolean;
 }
 
 export interface NormalizedTransformOptions extends TransformOptions {
     outputConfig: NormalizedOutputConfig;
     stylesheetConfig: NormalizedStylesheetConfig;
     experimentalDynamicComponent: NormalizedDynamicComponentConfig;
     isExplicitImport: boolean;
+    experimentalLightDOMComponent?: boolean;
```

Comment:
As far as I can tell, we only need to make this change in two places. (I'm not sure why two? But I see that `NormalizedTransformOptions` already duplicates some stuff from `TransformOptions`, so I just followed suit. If this is unnecessary, let me know.)

I'm also not sure if we need an options bag or not (to selectively enable light DOM slots? style scoping? I dunno.). So I just left it as a boolean.